draco/analysis/flagging.py

```diff
@@ -3144,6 +3144,176 @@ def process(self, rfimask):
+
+        Parameters
+        ----------
+        rfimask : containers.LocalizedRFIMask or containers.RFIMask
+            The original RFI mask to be realigned.
+
+        Returns
+        -------
+        out : containers.RFIMask or containers.LocalizedRFIMask
+            The RFI mask with its time axis aligned to the reference time axis.
+        """
+        # Ensure frequency axis is distributed
+        rfimask.redistribute("freq")
+
+        # Validate the mask data and extract the el axis
+        if isinstance(rfimask, containers.RFIMask):
+            el = None
+            nel = 1
+        elif isinstance(rfimask, containers.LocalizedRFIMask):
+            el = rfimask.el[:]
+            nel = len(el)
+        else:
+            raise ValueError(
+                f"Input class must be RFIMask or LocalizedRFIMask. Got {type(rfimask)}."
+            )
+
+        # Extract time and freq axes
+        freq = rfimask.freq[:]
+        mask_time = rfimask.time[:]
+        ntime = len(mask_time)
+
+        # Get the mask data
+        mask = rfimask.mask.local_data if rfimask.mask.distributed else rfimask.mask[:]
+
+        # The size of the local freq axis
+        nfreq_local = mask.shape[0]
+
+        # Reshape the mask so that the mask data has the same shape regardless its container type
+        mask = mask.reshape(nfreq_local, nel, ntime)
+
+        # Determine valid time range in the mask
+        start = np.abs(mask_time - self.target_time[0]).argmin()
+        end = np.abs(mask_time - self.target_time[-1]).argmin() + 1
+        if start == end:
+            raise ValueError(
+                "No target time values fall within the valid mask time range."
+            )
+
+        # Determine valid mask time window
+        valid_mask_range = slice(start, end)
+        mask_time_valid = mask_time[valid_mask_range]
+
+        # Nearest-interpolation: find the nearest index in target time for each input mask time
+        nearest_indices = np.abs(self.target_time[:, None] - mask_time_valid).argmin(
+            axis=0
+        )
+
+        # The time axis of the adjusted mask
+        valid_target_range = slice(nearest_indices[0], nearest_indices[-1] + 1)
+        new_time = self.target_time[valid_target_range]
+        nnew_time = len(new_time)
+
+        # Determine if conservative spreading can be skipped
+        mapped_times = mask_time[valid_mask_range]
+        if np.array_equal(mapped_times, new_time):
+            self.spread_size = 0
+
+        # Build index arrays for broadcasting
+        freq_idx = np.arange(nfreq_local)[:, None, None]
+        el_idx = np.arange(nel)[None, :, None]
+        new_time_idx = nearest_indices[None, None, :] - nearest_indices[0]
+
+        # Initialize and store adjusted mask
+        mask_adj = np.full((nfreq_local, nel, nnew_time), False)
+        np.logical_or.at(
+            mask_adj, (freq_idx, el_idx, new_time_idx), mask[:, :, valid_mask_range]
+        )
```

Comment:
```suggestion
    mask_adj[..., new_time_idx] |= mask[..., valid_mask_range]
```

This works, but it's a bit confusing because it isn't immediately clear that the `or` is being done in-place on `mask_adj`. Since you aren't making any selections along the `freq` or `el` axes, what I've suggested here should work fine while being easier to read. You would also be able to removed the lines defining `freq_idx` and `el_idx`.